### PR TITLE
Drop python 2.7 support

### PR DESCRIPTION
**Pull request by BeyondEvil**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,10 +8,6 @@ jobs:
       sudo: required
       env: TOXENV=linting
 
-    -
-      python: 2.7
```
Comment (nicoddemus) [inline .travis.yml:12]: Need to drop `pypy` test environment too.

**Comment by nicoddemus** [comment]
Strange, I don't quite understand the pypy3 failure... 🤔

**Comment by BeyondEvil** [comment]
> Strange, I don't quite understand the pypy3 failure... 🤔

Yeah, me neither. I'll investigate a little bit. If I can't figure it out on my own I'll raise the question with the pypy community.

**Comment by BeyondEvil** [comment]
It fails on (a copy of) [master branch](https://travis-ci.org/github/pytest-dev/pytest-base-url/builds/749481385), so seems unrelated to my changes. @nicoddemus 🤔 

Almost looks like the issue resides in pytester? 🧐

**Comment by nicoddemus** [comment]
Looks like it, but I can't phantom how pytest itself works (that bit of code is executed by basically every test that uses `pytester`/`testdir`).

**Pull request closed without merging**